src/main/java/io/nats/client/impl/SocketDataPort.java

```diff
@@ -187,4 +183,70 @@ protected static boolean isWebsocketScheme(String scheme) {
         return "ws".equalsIgnoreCase(scheme) ||
             "wss".equalsIgnoreCase(scheme);
     }
+
+    /**
+     * Implements the "Happy Eyeballs" algorithm as described in RFC 6555,
+     * which attempts to connect to multiple IP addresses in parallel to reduce
+     * connection setup delays.
+     */
+    private Socket connectToFastestIp(Options options, String hostname, int port,
+                                      int timeoutMillis) throws IOException {
+        // Get all IP addresses for the hostname
+        List<InetAddress> ips = Arrays.asList(InetAddress.getAllByName(hostname));
+
+        ExecutorService executor = Executors.newFixedThreadPool(ips.size());
+        long CONNECT_DELAY_MILLIS = 250;
```

Comment:
Why was 250 millis chosen? Should this be configurable?

Reply:
Got this from okhttp, other clients like Chrome, Firefox use 300ms. Not really needed to be configurable.

Reply:
Let's make this configurable, since you seem to have a reason to choose something other than 300 to match the browsers. Again, I feel like I'm make a collection of Android specific settings.